controller/server/main.go

```diff
@@ -138,6 +144,66 @@ func validatePath(path string) (string, error) {
 	return path, nil
 }
 
+func (s *server) CreateTopology(ctx context.Context, req *cpb.CreateTopologyRequest) (*cpb.CreateTopologyResponse, error) {
+	log.Infof("Received CreateTopology request: %v", req)
+	bp, err := validatePath(req.Topology.Name)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid topology path: %v", err)
+	}
+	p := topo.TopologyParams{
+		TopoName:       req.Topology.Name,
+		TopoNewOptions: []topo.Option{topo.WithBasePath(filepath.Dir(bp))},
+		Kubecfg:        kubecfg,
+	}
+	err = topo.CreateTopology(ctx, p)
+	if err != nil {
+		return &cpb.CreateTopologyResponse{
```

Comment:
err if you return an error here there is no contract on the object returned - if you want data included that the user can "rely on" you should return that as part of the error

Reply:
Done. Also revised CreateCluster() based on discussion with Alex.

For a finer granularity of error type, I will have another PR to propagate the error details from the library in topo.go.